hamilton/lifecycle/conditional_tqdm.py

```diff
@@ -0,0 +1,105 @@
+from typing import Any, Dict, List, Optional
+
+import tqdm
+
+from hamilton import graph, node
+from hamilton.lifecycle import base
+
+
+class TQDMHook(
+    base.BasePreGraphExecute,
+    base.BasePreNodeExecute,
+    base.BasePostNodeExecute,
+    base.BasePostGraphExecute,
+):
+    """A hook that uses tqdm to show progress bars for the graph execution.
```

Comment:
Lingo question -- I've been using adapter for a combination of hooks/methods. But this is just hooks so maybe its a hook?

Reply:
it goes in the adapter argument of the driver though.

Adapters have types is the way I see it.

Reply:
Simplified to just TQDM